### PR TITLE
Add gpu, cpu benchmark notebook demo and supporting files

### PR DESCRIPTION
**Pull request by skirui-source**
Fixes: https://github.com/rapidsai/deployment/issues/249

### CODE REVIEW

File: source/examples/xgboost-rf-gpu-cpu-benchmark/notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,353 @@
+{
```
Comment (hcho3) [inline source/examples/xgboost-rf-gpu-cpu-benchmark/notebook.ipynb:1]: <p>Can you format the commands using the Markdown code block? <a href="https://www.markdownguide.org/extended-syntax/#fenced-code-blocks" target="_blank">Use three backticks</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (hcho3) [inline source/examples/xgboost-rf-gpu-cpu-benchmark/notebook.ipynb:1]: <p>Format this cell (and other command cells) using the Markdown code block.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (hcho3) [inline source/examples/xgboost-rf-gpu-cpu-benchmark/notebook.ipynb:1]: <p>Format this cell (and other command cells) using the Markdown code block.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

**Review by hcho3 — APPROVED** [review]
Apart from the formatting, everything looks good to me.

**Comment by skirui-source** [comment]
@hcho3 addressed all your reviews, ready for another review. Unsure what is causing the failing CI tests

File: source/examples/xgboost-rf-gpu-cpu-benchmark/notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+   "id": "a51c95d1-b447-4f1b-9571-cf597ca93ef4",
+   "metadata": {},
+   "source": [
+    "<span style=\"display: block;  text-align: center; color:#8735fb; font-size:22pt\"> **HPO Benchmarking with RAPIDS and Dask** </span>\n",
+    "\n",
+    "Hyper-Parameter Optimization (HPO) helps to find the best version of a model by exploring the space of possible configurations. While generally desirable, this search is computationally expensive and time-consuming.\n",
+    "\n",
+    "In the notebook demo below, we compare benchmarking results to show how GPU can accelerate HPO tuning jobs relative to CPU.\n",
+    "\n",
+    "For instance, we find a x speedup in wall clock time and a x reduction in cost when comparing between GPU and CPU EC2 instances on 100 XGBoost HPO trials on 3 years of the Airline Dataset.\n"
```
Comment (hcho3) [inline source/examples/xgboost-rf-gpu-cpu-benchmark/notebook.ipynb:14]: Can you put in the actual performance numbers for c5.24xlarge and p3.8xlarge ?

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: source/examples/xgboost-rf-gpu-cpu-benchmark/notebook.ipynb (hunk 2 above)
Comment (skirui-source) [reply source/examples/xgboost-rf-gpu-cpu-benchmark/notebook.ipynb]: I will update with the latest results once the cpu benchmarks are done. Merging for now.

**Pull request merged**